src/mame/drivers/rainbow.cpp

```diff
@@ -1405,26 +1408,26 @@ WRITE_LINE_MEMBER(rainbow_state::mpsc_irq)
 WRITE8_MEMBER(rainbow_state::comm_bitrate_w)
 {
 	m_dbrg_A->str_w(data & 0x0f);  // PDF is wrong, low nibble is RECEIVE clock (verified in SETUP).
-	printf("\nRECEIVE bitrate = %02x HEX\n",data & 0x0f);
+	printf("\n(COMM.) receive bitrate = %d ($%02x)\n", comm_rates[data & 0x0f] , data & 0x0f);
```

Comment:
Could you please replace these `printf` statements with one of the MAME-provided things?  You have a choice of:
* Configurable logging from `logmacro.h` (preferred)
* Direct calls to `logerror`
* `osd_printf_verbose` etc. (least preferred as it doesn't automatically provide device context and isn't configurable)